### ISSUE TITLE
Should not warn on null default value on [AllowNull] parameter

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#40975]
Follow-up on https://github.com/dotnet/roslyn/pull/40789

```
        [Fact]
        public void AllowNull_Parameter_NullDefaultValue()
        {
            var source =
@"using System.Diagnostics.CodeAnalysis;
public class C
{
    void M([AllowNull] string p = null) { }
}
";

            var comp = CreateNullableCompilation(new[] { AllowNullAttributeDefinition, source });
            // Note: fixing this was tricky as it involved a cycle (FlowAnnotations, attribute decoding, default value)
            comp.VerifyDiagnostics(
                // (4,35): warning CS8625: Cannot convert null literal to non-nullable reference type.
                //     void M([AllowNull] string p = null) { }
                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(4, 35)
                );
        }
```

**Comment by RikkiGibson** [comment]
Will solve at the same time as #40818
